exporters/zipkin/src/test/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformerTest.java

```diff
@@ -149,19 +152,112 @@ void generateSpan_ResourceServiceNameMapping() {
   void generateSpan_defaultResourceServiceName() {
     SpanData data = spanBuilder().setResource(Resource.empty()).build();
 
-    Endpoint expectedEndpoint =
+    Endpoint expectedLocalEndpoint =
         Endpoint.newBuilder()
             .serviceName(Resource.getDefault().getAttribute(ResourceAttributes.SERVICE_NAME))
             .ip(localIp)
             .build();
     Span expectedZipkinSpan =
         zipkinSpan(Span.Kind.SERVER, localIp).toBuilder()
-            .localEndpoint(expectedEndpoint)
+            .localEndpoint(expectedLocalEndpoint)
             .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "OK")
             .build();
     assertThat(transformer.generateSpan(data)).isEqualTo(expectedZipkinSpan);
   }
 
+  @Test
+  void generateSpan_RemoteEndpointMapping() {
+    Attributes attributes =
+        Attributes.builder()
+            .put(PEER_SERVICE, "remote-test-service")
+            .put(NET_SOCK_PEER_ADDR, "8.8.8.8")
+            .put(NET_PEER_PORT, 42L)
+            .build();
+
+    SpanData spanData =
+        spanBuilder().setResource(Resource.empty()).setAttributes(attributes).build();
+
+    Endpoint expectedLocalEndpoint =
+        Endpoint.newBuilder()
+            .serviceName(Resource.getDefault().getAttribute(ResourceAttributes.SERVICE_NAME))
+            .ip(localIp)
+            .build();
+
+    Endpoint expectedRemoteEndpoint =
+        Endpoint.newBuilder().serviceName("remote-test-service").ip("8.8.8.8").port(42).build();
+
+    Span expectedSpan =
+        zipkinSpan(Span.Kind.SERVER, localIp).toBuilder()
+            .localEndpoint(expectedLocalEndpoint)
+            .remoteEndpoint(expectedRemoteEndpoint)
+            .putTag(PEER_SERVICE.getKey(), "remote-test-service")
+            .putTag(NET_SOCK_PEER_ADDR.getKey(), "8.8.8.8")
+            .putTag(NET_PEER_PORT.getKey(), "42")
```

Comment:
It's a bummer that these are kinda redundant (duplicated) now -- they exist both in the `remoteEndpoint` and in the tags. Should we filter these specific keys out of the tags when mapping the attributes over?

Reply:
Fyi: there will be lots of things to filter out: https://opentelemetry.io/docs/reference/specification/trace/sdk_exporters/zipkin/#otlp---zipkin

What happens in this class seems wrong to me (but I guess now it's too late for that); I think the span (`SpanData`?) should have these as separate fields not as attributes. I'm biased though: I prefer having type-safe fields instead of pouring everything into a "hashmap".

Also, just an fyi: as a user of OTel, I would consider this as a breaking change (a tag will be removed from the span). I'm not sure if doing such a change is even allowed here (maybe it is since semconv is unstable so deleting/renaming attributes and breaking users that way is possible at this point here).
It might worth to check other exporters too, i.e.: will those tags be shipped to Jaeger/OTLP/etc?

Because of these, I think deleting (filtering out) those attributes should happen in a separate issue/pr.

Reply:
> I'm biased though: I prefer having type-safe fields instead of pouring everything into a "hashmap".

I assume the decision to model fields in map-like structure instead of in top level fields is because it affords more ability to experiment without the consequences of adding a top level field we later wish to remove / rename / or change. This relieves bikeshedding (to some extent anyway 😄).

Additionally, since spans can potentially be used to model many many domains, the top level fields could easily get bogged down as more and more top level fields were added.

Reply:
>without the consequences of adding a top level field we later wish to remove / rename / or change

I think I see the intent but doing this will also give a false sense of not introducing a breaking change when this happens. Indeed, source and binary compatibility are kept but unfortunately it is still a breaking change since behavioral compatibility is not.

I totally agree with not adding fields for everything (that's what attributes are for) but I also think that this is such a fundamental field that it should be there (like the name of the span or the localEndpoint).

I just wanted to call these out, they don't really belong to the scope of this PR.

Reply:
It's a "fundamental" field for one type of span, but certainly not universally applicable, which is why it shouldn't be a top level field on the Span, or the SpanData.

Reply:
I guess that's why it should be optional/`@Nullable` but to me that does not make it less important.
The fact that it is optional/not universally applicable does not justifies "stringly" typing to me; if something is optional, it usually does not go into a hashmap.
I guess this is true for other parts of the OTel API/SDK, there are things that are optional but they have their own fields and they are not stringy typed, right?

Reply:
You're arguing with the wrong crowd here... this would be something that would need to be added in the specification, and not something we would or could unilaterally add to the java APIs.

Reply:
Yepp, that's why I wrote that I think now it's too late for that.